src/chainparams.cpp

```diff
@@ -240,11 +240,13 @@ static CTestNetParams testNetParams;
 /**
  * Regression test
  */
+const int DEFAULT_REGTEST_HALVING_INTERVAL = 150;
+
 class CRegTestParams : public CChainParams {
 public:
     CRegTestParams() {
         strNetworkID = "regtest";
-        consensus.nSubsidyHalvingInterval = 150;
+        consensus.nSubsidyHalvingInterval = DEFAULT_REGTEST_HALVING_INTERVAL;
```

Comment:
What speaks against directly use
```
consensus.nSubsidyHalvingInterval = GetArg("-halvinginterval", DEFAULT_REGTEST_HALVING_INTERVAL);
```

... and remove the `UpdateSubsidyHalvingIntervalParameter` function loop?